aptos-move/aptos-vm/src/counters.rs

```diff
@@ -78,30 +87,55 @@ pub static SYSTEM_TRANSACTIONS_EXECUTED: Lazy<IntCounter> = Lazy::new(|| {
     .unwrap()
 });
 
+const NUM_BLOCK_TRANSACTIONS_BUCKETS: [f64; 20] = [
+    5.0, 10.0, 20.0, 40.0, 75.0, 100.0, 125.0, 150.0, 175.0, 200.0, 250.0, 300.0, 350.0, 400.0,
+    450.0, 500.0, 750.0, 1000.0, 1500.0, 2000.0,
+];
+
 pub static BLOCK_TRANSACTION_COUNT: Lazy<Histogram> = Lazy::new(|| {
-    register_histogram!(
+    let histogram_opts = histogram_opts!(
         "aptos_vm_num_txns_per_block",
-        "Number of transactions per block"
-    )
-    .unwrap()
+        "Number of transactions per block",
+        NUM_BLOCK_TRANSACTIONS_BUCKETS.to_vec()
+    );
+    register_histogram!(histogram_opts).unwrap()
 });
 
+const TRANSACTION_EXECUTION_TIME_BUCKETS: [f64; 20] = [
+    0.002, 0.004, 0.008, 0.015, 0.02, 0.025, 0.03, 0.035, 0.04, 0.05, 0.06, 0.07, 0.08, 0.09, 0.10,
+    0.125, 0.15, 0.20, 0.40, 0.80,
+];
+
 pub static TXN_TOTAL_SECONDS: Lazy<Histogram> = Lazy::new(|| {
-    register_histogram!(
+    let histogram_opts = histogram_opts!(
         "aptos_vm_txn_total_seconds",
-        "Execution time per user transaction"
-    )
-    .unwrap()
+        "Execution time per user transaction",
+        TRANSACTION_EXECUTION_TIME_BUCKETS.to_vec()
+    );
+    register_histogram!(histogram_opts).unwrap()
 });
 
+const TRANSACTION_VALIDATION_TIME_BUCKETS: [f64; 14] = [
+    0.0005, 0.001, 0.0015, 0.002, 0.0025, 0.003, 0.0035, 0.004, 0.0045, 0.005, 0.006, 0.008, 0.01,
+    0.015,
+];
+
 pub static TXN_VALIDATION_SECONDS: Lazy<Histogram> = Lazy::new(|| {
-    register_histogram!(
+    let histogram_opts = histogram_opts!(
         "aptos_vm_txn_validation_seconds",
-        "Validation time per user transaction"
-    )
-    .unwrap()
+        "Validation time per user transaction",
+        TRANSACTION_VALIDATION_TIME_BUCKETS.to_vec()
+    );
+    register_histogram!(histogram_opts).unwrap()
 });
 
+const TXN_GAS_USAGE_BUCKETS: [f64; 10] = [2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0];
```

Comment:
Please check with move team on this. I believe we might need to accommodate higher gas usage. probably @vgao1996

Reply:
I think thats because the max bucket is 10 so all the values greater than 10 is counted in that last bucket. See [here](https://aptoslabs.grafana.net/explore?orgId=1&left=%7B%22datasource%22:%22fHo-R604z%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22fHo-R604z%22%7D,%22editorMode%22:%22code%22,%22expr%22:%22rate(aptos_vm_txn_gas_usage_sum%5B$__rate_interval%5D)/rate(aptos_vm_txn_gas_usage_count%5B$__rate_interval%5D)%22,%22legendFormat%22:%22__auto%22,%22range%22:true,%22instant%22:true%7D%5D,%22range%22:%7B%22from%22:%22now-1h%22,%22to%22:%22now%22%7D%7D) for average.

Reply:
Gotcha. I updated the bucket sizes accordingly.